lib/iris/_data_manager.py

```diff
@@ -0,0 +1,471 @@
+# (C) British Crown Copyright 2010 - 2017, Met Office
+#
+# This file is part of Iris.
+#
+# Iris is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Iris is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Iris.  If not, see <http://www.gnu.org/licenses/>.
+"""
+Management of common state and behaviour for cube and coordinate data.
+
+"""
+
+from __future__ import (absolute_import, division, print_function)
+from six.moves import (filter, input, map, range, zip)  # noqa
+import six
+
+import copy
+import numpy as np
+import numpy.ma as ma
+
+from iris._lazy_data import as_concrete_data, as_lazy_data, is_lazy_data
+from iris.util import array_equal
+
+
+class DataManager(object):
+    """
+    Provides a well defined API for management of real or lazy data.
+
+    """
+
+    def __init__(self, data, realised_dtype=None):
+        """
+        Create a data manager for the specified data.
+
+        Args:
+
+        * data:
+            The :class:`~numpy.ndarray` or :class:`~numpy.ma.core.MaskedArray`
+            real data, or :class:`~dask.array.core.Array` lazy data to be
+            managed.
+
+        Kwargs:
+
+        * realised_dtype:
+            The intended dtype of the specified lazy data.
```

Comment:
I think it might be worth stating this should only be set for non-float data. Else someone may do:
`DataManager(lazy_data, realised_dtype=np.float32)`
which would return 
`Can only cast lazy data to an integer or boolean dtype, got float.`